server/controllers/LibraryController.js

```diff
@@ -513,7 +513,7 @@ class LibraryController {
    */
   async getUserPlaylistsForLibrary(req, res) {
     let playlistsForUser = await Database.playlistModel.getPlaylistsForUserAndLibrary(req.user.id, req.library.id)
-    playlistsForUser = await Promise.all(playlistsForUser.map(async (p) => p.getOldJsonExpanded()))
+    playlistsForUser = await Promise.all(playlistsForUser.map(async (p) => p.getLimitedOldJsonExpanded()))
```

Comment:
Same thing as the collections endpoint, but looks like this function is written backwards for doing the paging at the database request level, so that may need to be swapped for the paging to actually work.

Reply:
To be clear, it was already requesting everything from the database and then limiting the response in memory, so maybe the `getUserPlaylistsForLibrary` function needs to be moved around

Reply:
Thanks @nichwall you are right. I made a mistake initially it looks like and we are loading all the items twice.
I'm going to start fresh on this one instead of working off this PR to clean more of this up

Reply:
This explains the slow loading actually because we are iterating through every item and loading the playlistMediaItems when we should be getting them all in one go.
It would be helpful to also add an optional query param to limit the items returned in each playlist so I'll look into that also

Reply:
Just updated this https://github.com/advplyr/audiobookshelf/commit/157616421841b2535d108e5c4d402931429d4075
I wasn't able to limit the media items in that query because of a sequelize bug https://github.com/sequelize/sequelize/issues/12971
The only other way I can think to implement that is to filter out media items with an `order` < 5 or something like that.
It would be better to migrate that endpoint to return the new data model then to spend a lot more time trying to get the old one to perform better.